plugins/woocommerce-blocks/assets/js/base/context/hooks/cart/use-store-cart-coupons.ts

```diff
@@ -41,6 +45,23 @@ export const useStoreCartCoupons = ( context = '' ): StoreCartCoupon => {
 		);
 
 	const { applyCoupon, removeCoupon } = useDispatch( CART_STORE_KEY );
+	const orderId = useSelect( ( select ) =>
+		select( CHECKOUT_STORE_KEY ).getOrderId()
+	);
+
+	// Check if we're on the cart or checkout page and set the error message accordingly.
+	const getCouponErrorMessage = ( error: ApiErrorResponse ) => {
+		// Set the Checkout error message.
+		if ( orderId && orderId > 0 && error?.data?.details?.checkout ) {
+			return error.data.details.checkout;
+		}
+		// Set the Cart error message.
+		else if ( error?.data?.details?.cart ) {
+			return error.data.details.cart;
+		}
+
+		return error.message;
+	};
```

Comment:
The comments look a bit redundant. What are your thoughts on adjusting the main comment and removing the inner comments? This is what I had in mind:

```suggestion
	// Return cart, checkout or generic error message.
	const getCouponErrorMessage = ( error: ApiErrorResponse ) => {
		if ( orderId && orderId > 0 && error?.data?.details?.checkout ) {
			return error.data.details.checkout;
		} else if ( error?.data?.details?.cart ) {
			return error.data.details.cart;
		}

		return error.message;
	};
```

Reply:
Good catch, makes sense. Too many words there 👍🏼